### DISCUSSION TITLE
Line charts are missing lines

### DISCUSSION THREAD

**Discussion by mmays77** [Budibase/budibase#14580] [Help]
Budibase community version 2.32.3 self hosted Ubuntu 22.04
I have not been able to find any mention of this issue.

I have a number of python rest queries that provide pivot tables. The data check out. The data are loading into tables in Budibase fine. When I add a line chart, sometimes the data consistently display just fine. Some line charts, though, are missing lines. I have charts that are missing all of the lines. If I reduce the number of data columns, or the x-axis dates displayed using filters in the data provider, some of the lines show up. The interesting thing is that when I hover over the chart, the data value bubbles appear showing the correct numeric values, even though the lines are missing. It seems like it might be running out of memory or something, but I only have about 18 x axis data points, and lose lines sometimes with only 3 data columns selected.
![BudibaseChartExample](https://github.com/user-attachments/assets/5b11dc2a-6e21-4d12-8584-1cde74702004)

**Comment by mikesealey** [comment]
Hey @mmays77 this looks like a bug - could you fill out the bug report template  in the "issues" section, giving as much detail as possible.

**Comment by mmays77** [comment]
Thank you for your reply. 

At this point, I cannot reproduce the problem, because by rebuilding the screen and data objects, the problems went away. I believe that my problem is related to changes in the data columns returned by a rest call. This has happened several times.

When I add or change columns in a rest service call, and the data object in the rest design area in BudiBase does not easily update the schema. I have tried a number of things and I have no clear process for updating the schema. I rerun the rest call, and old columns appear in the schema. I've tried to remove all the columns from the rest object in design, refreshed screens, and sometimes it takes several tries to get the schema represented correctly. On one occasion I had to recreate the rest call in BudiBase from scratch. It would be nice to have an unambiguous process that will update the schema with new columns and delete columns every time.

Data providers are also difficult to get to reflect the changed schema. Sometimes the only way to get a data provider to reflect a changed schema for a rest service call is to build a new one from scratch, adding both the data provider and the table as new objects. Creating new line charts from scratch fixes those related issues, as well.

Sorry for the long-winded explanation. Overall I'm happy with my BudiBase app. The issue isn't a high level problem for me, but it is frustrating that I have to go through a Kabuki dance to update a rest service and the screen objects that depend on it by having to rebuild everything from scratch.
